python/xorbits/_mars/dataframe/groupby/aggregation.py

```diff
@@ -1017,7 +1017,14 @@ def _do_predefined_agg(input_obj, agg_func, single_func=False, **kwds):
 
         if ndim == 2:
             if single_func:
-                result = input_obj.agg(agg_func)
+                # DataFrameGroupby.agg('size') returns empty df in cudf, which is not correct
+                # The index order of .size() is wrong in cudf,
+                # however, for performance considerations, sort_index() will not be called here
```

Comment:
> `sort_index` is not needed since the dataframe will be reindexed later, right?

Actually, when there is only one chunk, if the order of the results needs to be consistent with pandas, ``sort_index`` needs to be called because the order of cudf calculation results itself is problematic. But ``sort_index``  is not efficient. So for this method we can only keep the values ​​correct, but the order can only depend on the result of cudf.